server/api/mission.js

```diff
@@ -250,6 +250,65 @@ router.get('/mission/list/history/:id/bonus', jwtAuth, async (req, res, next) =>
   }
 });
 
+router.get('/mission/:missionId/user/:userId', jwtAuth, async (req, res, next) => {
+  try {
+    const { missionId, userId } = req.params;
+    const { userMissionList = [] } = req.query;
+
+    const [missionCol, userMission] = await Promise.all([
+      missionsRef.get(),
+      dbRef.doc(userId).collection('mission').doc(missionId).get(),
+    ]);
+
+    const getMission = id => missionCol.docs.find(d => d.id === id);
```

Comment:
please add a comment stating that you need to trace back required mission chain, so need to fetch the whole doc instead of query